### DISCUSSION TITLE
sysext for k3s or rke2?

### DISCUSSION THREAD

**Discussion by kastl-ars** [flatcar/Flatcar#1439] [Ideas]
Hi all,

I just read the docs on the "new" sysext capabilities. I found one issue on packaging Podman as a sysext.

Has anyone thought of having sysexts for k3s or rke2 kubernetes distributions that come with (more or less) a single binary?

* https://k3s.io/
* https://docs.rke2.io/

Especially k3s is really nice on limited hardware, so combining this with Flatcar would be really interesting.

Kind Regards,
Johannes

**Comment by ader1990** [comment]
Hello @kastl-ars,

K3s systemd-sysext support has been just added a few days ago: https://github.com/flatcar/sysext-bakery/pull/53.

**Reply by kastl-ars** [reply]
Thanks, nice to hear that. 
That sysext just contains the binary, are the service files in a confext or are they left out intentionally?

If I find time on the weekend, I could come up with a similar script for rke2...

**Reply by tormath1** [reply]
@kastl-ars confext are not yet tested / supported at the moment on Flatcar. IIRC `k3s` takes care of installing his own systemd units? If you think of configuration inside `/etc` you can provide additional units that takes care of linking / copying files from /usr to /etc or use systemd-tmpfiles (e.g: https://github.com/flatcar/scripts/blob/59842e9ee706a039d708cb655d285fecba20bca4/build_library/sysext_mangle_flatcar-zfs#L30-L36)

**Reply by kastl-ars** [reply]
> IIRC k3s takes care of installing his own systemd units?

Not as far as I know. The script that can be used to install/download k3s can do that.

Regarding the confext, I thought that for some reason the unit files need to go to /etc/ and thus were omitted from the sysext. Normally they should be in `/usr/lib/systemd/system/`.

**Reply by kastl-ars** [reply]
I opened a PR to add rke2: https://github.com/flatcar/sysext-bakery/pull/68

And I opened another PR to add the k3s service files: https://github.com/flatcar/sysext-bakery/pull/69

Both are using `/usr/bin/local` to a) match upstream and b) separate sys-ext installed binaries from system binaries.
